PostgresAssignment/src/controller/class.ts

```diff
@@ -0,0 +1,67 @@
+import { Context } from "vm";
+import {
+	getClasses as get_classes,
+	studentOfClass as student_of_classes,
+	addClass as add_class,
+} from "../services/class";
+import { classSchema } from "../helper/validation";
+
+interface classI {
+	name: string;
+	room: number;
+}
+
+export async function getClass(ctx: Context): Promise<void> {
+	try {
+		const allClasses = await get_classes();
+		ctx.response.status = 200;
+		ctx.body = {
+			count: allClasses.length,
+			data: allClasses,
+		};
+	} catch (e) {
+		ctx.status = 500;
+		if (e.status) ctx.status = e.status;
+		ctx.body = { error: e.message };
+	}
+}
+
+export async function studentsOfClass(ctx: Context): Promise<void> {
+	const classId: string = ctx.params.classId;
+	try {
+		const requiredStudent = await student_of_classes(classId);
+		ctx.response.status = 200;
+		ctx.body = {
+			msg: "required student detail",
+			data: requiredStudent,
+		};
+	} catch (e) {
+		ctx.status = 500;
+
+		if (e.status) ctx.status = e.status;
+		ctx.body = { error: e.message };
+	}
+}
+
+export async function addClass(ctx: Context): Promise<void> {
+	const obj: classI = ctx.request.body;
+	try {
+		const response = await classSchema.validate(obj);
```

Comment:
Can we please move this outside try block ? I don't think this will throw exception.